# data loading and understanding

In [1]:
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

df = pd.read_csv('/kaggle/input/student-academic-stress-real-world-dataset/academicStress.csv')

In [2]:
print("Dataset Shape:", df.shape)


Dataset Shape: (121, 9)


In [3]:
print(df.columns.tolist())


['Timestamp', 'Your Academic Stage', 'Peer pressure', 'Academic pressure from your home', 'Study Environment', 'What coping strategy you use as a student?', 'Do you have any bad habits like smoking, drinking on a daily basis?', 'What would you rate the academic  competition in your student life', 'Rate your academic stress index ']


In [4]:
print(df.info())


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 121 entries, 0 to 120
Data columns (total 9 columns):
 #   Column                                                               Non-Null Count  Dtype 
---  ------                                                               --------------  ----- 
 0   Timestamp                                                            121 non-null    object
 1   Your Academic Stage                                                  121 non-null    object
 2   Peer pressure                                                        121 non-null    int64 
 3   Academic pressure from your home                                     121 non-null    int64 
 4   Study Environment                                                    121 non-null    object
 5   What coping strategy you use as a student?                           121 non-null    object
 6   Do you have any bad habits like smoking, drinking on a daily basis?  121 non-null    object
 7   What would you rat

In [5]:
df.head(10)

Timestamp Your Academic Stage  Peer pressure  \
0  24/07/2025 22:05:39       undergraduate              4   
1  24/07/2025 22:05:52       undergraduate              3   
2  24/07/2025 22:06:39       undergraduate              1   
3  24/07/2025 22:06:45       undergraduate              3   
4  24/07/2025 22:08:06       undergraduate              3   
5  24/07/2025 22:08:13       undergraduate              3   
6  24/07/2025 22:09:21       undergraduate              5   
7  24/07/2025 22:10:06       undergraduate              3   
8  24/07/2025 22:11:01       undergraduate              2   
9  24/07/2025 22:11:19       undergraduate              2   

   Academic pressure from your home Study Environment  \
0                                 5             Noisy   
1                                 4          Peaceful   
2                                 1          Peaceful   
3                                 2          Peaceful   
4                                 3          Peaceful   
5                                 3          Peaceful   
6                                 5         disrupted   
7                                 2          Peaceful   
8                                 2          Peaceful   
9                                 2          Peaceful   

          What coping strategy you use as a student?  \
0  Analyze the situation and handle it with intel...   
1  Analyze the situation and handle it with intel...   
2                   Social support (friends, family)   
3  Analyze the situation and handle it with intel...   
4  Analyze the situation and handle it with intel...   
5  Analyze the situation and handle it with intel...   
6                 Emotional breakdown (crying a lot)   
7                   Social support (friends, family)   
8  Analyze the situation and handle it with intel...   
9  Analyze the situation and handle it with intel...   

  Do you have any bad habits like smoking, drinking on a daily basis?  \
0                                                 No                    
1                                                 No                    
2                                                 No                    
3                                                 No                    
4                                                 No                    
5                                                 No                    
6                                                 No                    
7                                                 No                    
8                                                 No                    
9                                                 No                    

   What would you rate the academic  competition in your student life  \
0                                                  3                    
1                                                  3                    
2                                                  2                    
3                                                  4                    
4                                                  4                    
5                                                  4                    
6                                                  4                    
7                                                  3                    
8                                                  2                    
9                                                  4                    

   Rate your academic stress index   
0                                 5  
1                                 3  
2                                 4  
3                                 3  
4                                 5  
5                                 4  
6                                 4  
7                                 3  
8                                 2  
9                                 2

In [6]:
print("Statistical Summary for Numeric Columns:")
df.describe()

Statistical Summary for Numeric Columns:


Peer pressure  Academic pressure from your home  \
count     121.000000                        121.000000   
mean        3.008264                          3.190083   
std         1.083942                          1.260384   
min         1.000000                          1.000000   
25%         2.000000                          2.000000   
50%         3.000000                          3.000000   
75%         4.000000                          4.000000   
max         5.000000                          5.000000   

       What would you rate the academic  competition in your student life  \
count                                         121.000000                    
mean                                            3.553719                    
std                                             0.991215                    
min                                             1.000000                    
25%                                             3.000000                    
50%                                             4.000000                    
75%                                             4.000000                    
max                                             5.000000                    

       Rate your academic stress index   
count                        121.000000  
mean                           3.669421  
std                            1.059783  
min                            1.000000  
25%                            3.000000  
50%                            4.000000  
75%                            4.000000  
max                            5.000000

In [7]:
# For categorical columns
categorical_cols = df.select_dtypes(include=['object']).columns
if len(categorical_cols) > 0:
    print("Unique Values per Categorical Column:")
    for col in categorical_cols:
        print(f"{col}: {df[col].nunique()} unique values")

Unique Values per Categorical Column:
Timestamp: 121 unique values
Your Academic Stage: 3 unique values
Study Environment: 3 unique values
What coping strategy you use as a student?: 3 unique values
Do you have any bad habits like smoking, drinking on a daily basis?: 3 unique values


* Dataset size: 121 rows × 9 columns — small enough for EDA but enough variation for analysis.
* Balanced mix: 4 numeric columns (peer-pressure, academic-pressure-home, academic-competition, academic-stress-index) and 5 categorical columns.
* Peer Pressure: Mean ≈ 3.01, Std ≈ 1.08 — moderately spread, values range 1–5.
* Academic Pressure from Home: Mean ≈ 3.19 — slightly higher than peer pressure.
* Academic Competition Rating: Mean ≈ 3.55 — most students feel competition is above average.
* Academic Stress Index: Mean ≈ 3.67 — stress levels lean toward the higher side.
* Your Academic Stage: 3 unique values — likely undergraduate/postgraduate/other group.
* Study Environment: only 3 types (e.g., Peaceful, Noisy, Disrupted).
* Coping Strategy: 3 common strategies — possibly a focus area for mental health recommendations.
* Bad Habits: Yes/No/Other — could test correlation with stress index.
* Higher academic competition seems linked to higher stress index.
* Poorer study environment (Noisy/Disrupted) may raise stress.
* Coping strategy effectiveness could be analyzed by comparing average stress index.
